### PR TITLE
Display Ingress host as a link

### LINKED ISSUE #1431: Ingress hosts field should be a link
On the Ingress summary page, inside the table containing all known ingresses, the `Hosts` column should be clickable. Currently we have it as a label:

Screen Shot 2020-09-29 at 1 28 11 PM

### PR DESCRIPTION
**Pull request by mklanjsek**
Instead of showing plain text, we will now try to detect if ingress host field contains the Url, and show it as a link if it does. 

This change also modifies the Link component to open the **external** links inside the separate browser window. This makes sense especially with a move to Electron app. All internal Octant links are handled same as before.

**Which issue(s) this PR fixes**
- Fixes #1431

### CODE REVIEW

File: web/src/app/modules/shared/components/presentation/link/link.component.html [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <ng-template [ngIf]="isExternal" [ngIfElse]="internal">
-  <a href="{{ ref }}">{{ value }}</a>
+  <a target ="_blank" href="{{ ref }}">{{ value }}</a>
```
Comment (wwitzel3) [inline web/src/app/modules/shared/components/presentation/link/link.component.html:2]: Let's make sure we test this with a plugin and make sure that changing the target here doesn't impact the expected behavior.

Comment (mklanjsek) [reply web/src/app/modules/shared/components/presentation/link/link.component.html:2]: Verified that both internal and external links work as expected inside my test plugin. Also tested that all links are working as expected inside existing plugins `knative` and `Jenkins X`

**Comment by mklanjsek** [comment]
@wwitzel3 rebased and ready to go

File: internal/printer/ingress.go [hunk 2]
```diff
@@ -54,6 +55,15 @@ func IngressListHandler(ctx context.Context, list *extv1beta1.IngressList, optio
 	return ot.ToComponent()
 }
 
+func getHostComponent(link string) component.Component {
+	_, err := url.Parse(link)
+	if strings.Contains(link, ".") && err == nil {
+		return component.NewLink("", link, "http://"+link)
```
Comment (wwitzel3) [inline internal/printer/ingress.go:61]: Can we look at the Ingress and determine if it is TLS/SSL?

Comment (wwitzel3) [reply internal/printer/ingress.go:61]: @mklanjsek ping

Comment (mklanjsek) [reply internal/printer/ingress.go]: Added TLS handling, also cases when multiple Urls are present.

**Comment by mklanjsek** [comment]
@wwitzel3 PTAL

**Review by wwitzel3 — CHANGES_REQUESTED** [review]
This is good, need to add some test cases to internal/printer/ingress_test.go 
There is already some decent test cases there to copy and extend.

**Comment by mklanjsek** [comment]
@wwitzel3 good point, added tests around this functionality.

**Review by wwitzel3 — APPROVED** [review]

**Pull request merged**